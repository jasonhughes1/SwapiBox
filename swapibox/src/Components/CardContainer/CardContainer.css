.cards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center; }

.select-favs {
  color: white; }
