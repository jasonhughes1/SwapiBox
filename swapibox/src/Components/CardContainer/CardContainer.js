import React from 'react';
import Card from './../Card/Card.js';
import './CardContainer.css';

const CardContainer = ({ cardType, setFavorite }) => {
  if (cardType.length > 0) {
    const cards = cardType.map((card, index) =>
      <Card cardData={card}
                 key={index}
                 setFavorite={setFavorite}
       />
    );

    return (
      <div className='card-container'>
        <div className='cards'>{cards}</div>
      </div>
    );
  } else {
    return (
      <h2 className='select-favs'>There are no favorites!</h2>
    )
  }
};



export default CardContainer;
