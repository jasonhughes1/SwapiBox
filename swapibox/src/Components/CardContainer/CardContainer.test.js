import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

let mockCardType = [{
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}, {
  Name: "Luke Skywalker",
  Homeworld: "Tatooine",
  Species: "Human",
  Population: "200000"
}];

let mockFavArray = [{
  Homeworld: "Tatooine",
  Name: "C-3PO",
  Population: "200000",
  Species: "Droid"
}];

let mockSetFavorite = jest.fn();
let renderedCardContainer;

describe('CardContainer Tests', () => {
  renderedCardContainer = shallow(
    <CardContainer
      cardType={mockCardType}
      setFavorite={mockSetFavorite}
      mockFavArray={mockFavArray}
    />
  );

  it('should render correctly', () => {
    expect(renderedCardContainer).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(renderedCardContainer).toMatchSnapshot();
  });
});
