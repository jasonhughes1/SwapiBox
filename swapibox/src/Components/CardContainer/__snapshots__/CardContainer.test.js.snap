// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer                   Tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardType={
        Array [
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
          ]
    }
    setFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cards"
>
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
        <Card
                cardData={
                        Object {
                                "Homeworld": "Tatooine",
                                "Name": "Luke Skywalker",
                                "Population": "200000",
                                "Species": "Human",
                              }
                }
                setFavorite={[Function]}
        />
</div>,
      "className": "card-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
          <Card
            cardData={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Name": "Luke Skywalker",
                                    "Population": "200000",
                                    "Species": "Human",
                                  }
            }
            setFavorite={[Function]}
/>,
        ],
        "className": "cards",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "Homeworld": "Tatooine",
              "Name": "Luke Skywalker",
              "Population": "200000",
              "Species": "Human",
            },
            "setFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cards"
>
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
          <Card
                    cardData={
                              Object {
                                        "Homeworld": "Tatooine",
                                        "Name": "Luke Skywalker",
                                        "Population": "200000",
                                        "Species": "Human",
                                      }
                    }
                    setFavorite={[Function]}
          />
</div>,
        "className": "card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
            <Card
              cardData={
                            Object {
                                          "Homeworld": "Tatooine",
                                          "Name": "Luke Skywalker",
                                          "Population": "200000",
                                          "Species": "Human",
                                        }
              }
              setFavorite={[Function]}
/>,
          ],
          "className": "cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "function",
            "props": Object {
              "cardData": Object {
                "Homeworld": "Tatooine",
                "Name": "Luke Skywalker",
                "Population": "200000",
                "Species": "Human",
              },
              "setFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
